src/main/catalog-sources/kubeconfig-sync.ts

```diff
@@ -156,14 +156,13 @@ export function computeDiff(contents: string, source: RootSource, port: number,
       for (const [contextName, model] of models) {
         // add new clusters to the source
         try {
-          const cluster = new Cluster({ ...model, id: uuid.v4() });
+          const clusterId = createHash("md5").update(`${filePath}:${contextName}`).digest("hex");
+          const cluster = ClusterStore.getInstance().getById(clusterId) || new Cluster({ ...model, id: clusterId});
```

Comment:
Uses saved cluster for base if it exists. Otherwise we don't see any saved changes on catalog items.

Reply:
If `clusterId` already exists we shouldn't even be in this code path...